src/main/resources/static/transactionHistory.html

```diff
@@ -31,18 +45,46 @@ <h1>Transaction History</h1>
     }
 
     async function filterByAmount() {
-        await fetch("/history/amount", {
+        let min = document.getElementById("minAmount").value;
+        let max = document.getElementById("maxAmount").value;
+        let request = '/history/amount?minAmount=' + min + '&maxAmount=' + max
+        await fetch(request, {
             method: "GET"
         })
             .then(response => response.json())
             .then(response => transactionsFromResponse = response)
 
+        while (transactionsList.firstChild)
+            transactionsList.removeChild(transactionsList.firstChild);
+
         transactionsFromResponse.forEach(transaction => {
```

Comment:
Just a FYI: an async function is required only when modifying "global" variables as you did here (`.then(response => transactionsFromResponse = response)`). If you want to avoid async you could just process the `transactionsFromResponse` in the `.then` callback instead of setting that value in a variable.

Reply:
Since you have 2 functions that use the same `transactionsFromResponse` variable I think that is a good ideea to declare this variable inside each function's scope (a local var inside the function `{...}`).